src/api/PDFDocument.ts

```diff
@@ -713,6 +713,29 @@ export default class PDFDocument {
     return copiedPages;
   }
 
+  /**
+   * Add document JavaScript. The script is executed when the document is opened.
+   * See the [JavaScript™ for Acrobat® API Reference](https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/js_api_reference.pdf)
+   * for details.
+   * @param script The JavaScript to execute.
+   * @param name The name of the script. Must be unique per document.
+   */
+  addJavascript(script: string, name: string) {
+    const jsActionDict = this.context.obj({
+      Type: 'Action',
+      S: 'JavaScript',
+      JS: PDFHexString.fromText(script),
+    });
+    const jsActionRef = this.context.register(jsActionDict);
+    const jsNameTree = this.context.obj({
+      Names: this.context.obj([name, jsActionRef]),
+    });
+    this.catalog.set(
```

Comment:
It looks like this will override any existing JavaScript that may exist in the document. We need to add to the existing tree if one exists, otherwise we should create a new one.

Reply:
Hi! Thanks for the quick feedback and warm welcome! 😊 This should be fixed by the new commits.